File: my-solutions/0026-remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int curr = nums[0];

        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] == curr) {
                nums.erase(nums.begin() + i);
                --i;
            } else {
                curr = nums[i];
            }
        }

        return nums.size();
    }
};

File: my-solutions/0026-remove-duplicates-from-sorted-array/solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        swap, search, count = 0, 1, 1
        while search < len(nums):
            if nums[search] != nums[swap]:
                nums[search], nums[swap + 1] = nums[swap + 1], nums[search]
                count += 1
                swap += 1
            
            search += 1
        
        return count
